games-emulation/jrommanager/jrommanager-1.6-r1.ebuild

```diff
@@ -0,0 +1,43 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+MY_PN=JRomManager
+MY_PV=1.6b20
+MY_P="${MY_PN}-${MY_PV}"
+
+inherit desktop
+
+DESCRIPTION="A Mame and Retrogaming Rom Manager, Java alternative to ClrMamePro"
+HOMEPAGE="https://github.com/optyfr/JRomManager"
+SRC_URI="https://github.com/optyfr/${MY_PN}/releases/download/${MY_PV}/${MY_P}.zip -> ${P}.zip"
+QA_PREBUILT="*"
+
+S="${WORKDIR}"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND=">=virtual/jre-1.8"
+DEPEND="${DEPEND}
+	app-arch/unzip"
+
+src_prepare() {
+	default
+
+	sed -e "/dir=.*/s:.*:cd ${EROOT}/usr/share/${PN}:" -i *.sh || die "sed failed!"
+}
+
+src_install() {
+	insinto "/usr/share/${PN}"
+	doins *.jar
+	doins -r lib
+	exeinto "/usr/share/${PN}"
+	doexe *.sh
+	dosym "${EROOT}/usr/share/${PN}/JRomManager-multi.sh" "/usr/bin/${PN}"
+	doicon "${FILESDIR}/${PN}.png"
+	make_desktop_entry "${PN}" '' '' 'Utility'
+	domenu jrommanager
```

Comment:
Doesn't the filename need to end with `.desktop`?

Reply:
Well surprisingly no... it works without

Reply:
Are you sure? Because my guess is, it just fails silently because of the bad logic inside the eclass.

Reply:
It created a valid entry in the menu (seen it under my xfce desktop)

Reply:
No. `make_desktop_entry` created a valid entry in the menu. `domenu` didn't do anything. Read the code. This is not magic, things don't happen unless they're programmed to happen.

Reply:
That's my guess.

Reply:
OK, so my conclusion is that domenu (or newmenu) are useless when creating .desktop with make_desktop_entry